src/sys/socket/ffi.rs

```diff
@@ -23,7 +23,10 @@ pub type type_of_cmsg_len = socklen_t;
 #[cfg(target_os = "macos")]
```

Comment:
I'd actually suggest putting these in a `cfg_if!` macro, as this will be a lot easier to understand.